### PR TITLE
feat(#1312): early dataset name validation

### LINKED ISSUE #1312: [Logging] Validate dataset name when rb.log
Try to validate the dataset name early and raise an well defined error if name fails validation

Closes #1200

### PR DESCRIPTION
**Pull request by frascuchon**
Closes #1312

### CODE REVIEW

File: src/rubrix/client/api.py [hunk 1]
```diff
@@ -213,6 +217,12 @@ def log(
         if not name:
             raise InputValueError("Empty project name has been passed as argument.")
```
Comment (dvsrepo) [inline src/rubrix/client/api.py]: Not from this PR but shouldn't be dataset name?

File: src/rubrix/client/api.py [hunk 2]
```diff
@@ -213,6 +217,12 @@ def log(
...
 
+        if not re.match(DATASET_NAME_REGEX_PATTERN, name):
+            raise InputValueError(
+                f"Provided name {name} does not match the pattern {DATASET_NAME_REGEX_PATTERN}. "
```
Comment (dvsrepo) [inline src/rubrix/client/api.py]:
```suggestion
                f"Provided dataset name {name} does not match the pattern {DATASET_NAME_REGEX_PATTERN}. "
```

**Review by dvsrepo — CHANGES_REQUESTED** [review]

File: src/rubrix/client/api.py (hunk 1 above)
Comment (frascuchon) [reply src/rubrix/client/api.py]: Sep

Comment (frascuchon) [inline src/rubrix/client/api.py]:
```suggestion
            raise InputValueError("Empty dataset name has been passed as argument.")
```

Comment (frascuchon) [reply src/rubrix/client/api.py]: Done

**Review by dvsrepo — APPROVED** [review]

**Pull request merged**